### PR TITLE
Search box API

### LINKED ISSUE #93: implement/activate ontology search box functionality of i2b2
this issues is complementary to this one https://github.com/ldsec/glowing-bear-medco/issues/148 in the front-end

### PR DESCRIPTION
**Pull request by f-marino**

### CODE REVIEW

File: connector/server/handlers/medco_node.go [hunk 1]
```diff
@@ -16,6 +16,34 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// MedCoNodeExploreSearchHandler handles the /medco/node/explore/search API endpoint.
+func MedCoNodeExploreSearchHandler(params medco_node.ExploreSearchParams, principal *models.User) middleware.Responder {
+
+	// set default value for limit if not present
+	if params.SearchRequest.Limit == nil {
+		params.SearchRequest.Limit = new(int64)
+		*params.SearchRequest.Limit = 10
```
Comment (nfreundl) [inline connector/server/handlers/medco_node.go]: Would it be better to set default values in swagger definition YAML ?

File: connector/server/handlers/medco_node.go [hunk 2]
```diff
@@ -16,6 +16,34 @@ import (
...
+		logrus.Debugf("invalid limit, defaulted: %d", *params.SearchRequest.Limit)
+	}
+
+	logrus.Infof("received request at /medco/node/explore/search with parameters: %v %v", *params.SearchRequest.SearchString, *params.SearchRequest.Limit)
```
Comment (nfreundl) [inline connector/server/handlers/medco_node.go]: received request at /medco/node/explore/search with parameters: %s %d

File: connector/server/handlers/medco_node.go [hunk 3]
```diff
@@ -16,6 +16,34 @@ import (
...
+
+	var searchResult []*models.ExploreSearchResultElement
+
+	searchResult, err := i2b2.GetOntologyElements(*params.SearchRequest.SearchString, *params.SearchRequest.Limit)
+	if err != nil {
+		return medco_node.NewExploreSearchDefault(500).WithPayload(&medco_node.ExploreSearchDefaultBody{
+			Message: err.Error(),
```
Comment (nfreundl) [inline connector/server/handlers/medco_node.go]: Would add more descriptive comment like: 
`Message:  "Search execution error:" + err.Error() `

**Review by nfreundl — CHANGES_REQUESTED** [review]
Only a few details to be changed in one file

File: connector/server/handlers/medco_node.go (hunk 1 above)
Comment (f-marino) [reply connector/server/handlers/medco_node.go]: It is set, but default values do not work with nested properties in go swagger, that's why I put this here. But I found a nicer workaround.

**Comment by f-marino** [comment]
Updated

**Review by nfreundl — APPROVED** [review]

**Pull request merged**